test/external/druidExternalIntrospection.mocha.coffee

```diff
@@ -0,0 +1,525 @@
+        "intervals": null,
+        "size": 0,
+        "numRows": 654321,
+        "columns": {
+          "__time": { "type": "LONG", "hasMultipleValues": false, "size": 0, "cardinality": null, "errorMessage": null },
+          "added": { "type": "FLOAT", "hasMultipleValues": false, "size": 0, "cardinality": null, "errorMessage": null },
+          "anonymous": { "type": "STRING", "hasMultipleValues": false, "size": 0, "cardinality": 0, "errorMessage": null },
+          "count": { "type": "LONG", "hasMultipleValues": false, "size": 0, "cardinality": null, "errorMessage": null },
+          "delta_hist": { "type": "approximateHistogram", "hasMultipleValues": false, "size": 0, "cardinality": null, "errorMessage": null },
+          "language": { "type": "STRING", "hasMultipleValues": false, "size": 0, "cardinality": 0, "errorMessage": null },
+          "namespace": { "type": "STRING", "hasMultipleValues": true, "size": 0, "cardinality": 0, "errorMessage": null },
+          "newPage": { "type": "STRING", "hasMultipleValues": false, "size": 0, "cardinality": 0, "errorMessage": "lol wtf" },
+          "page": { "type": "STRING", "hasMultipleValues": false, "size": 0, "cardinality": 0, "errorMessage": null },
+          "time": { "type": "STRING", "hasMultipleValues": false, "size": 0, "cardinality": 0, "errorMessage": null },
+          "user_unique": { "type": "hyperUnique", "hasMultipleValues": false, "size": 0, "cardinality": null, "errorMessage": null }
+        }
+      }
+
+      if query.analysisTypes.indexOf("aggregators") isnt -1
+        merged.aggregators = {
+          "added": { "type": "doubleSum", "name": "added", "fieldName": "added" },
+          "count": { "type": "longSum", "name": "count", "fieldName": "count" },
+          "delta": { "type": "doubleSum", "name": "delta", "fieldName": "delta" },
+          "delta_hist": { "type": "approxHistogramFold", "name": "delta_hist", "fieldName": "delta_hist", "resolution": 50, "numBuckets": 7, "lowerLimit": "-Infinity", "upperLimit": "Infinity" },
+          "user_unique": { "type": "hyperUnique", "name": "user_unique", "fieldName": "user_unique" }
+        }
+
+      return Q([merged])
+
+    else if query.queryType is 'introspect'
+      return Q({
+        dimensions: ['anonymous', 'language', 'namespace', 'newPage', 'page', 'time']
+        metrics: ['added', 'count', 'delta_hist', 'user_unique']
+      })
+
+    else
+      throw new Error('non introspection query')
+
+
+  requesterDruid_0_8_3 = ({query}) ->
+    expect(query.dataSource).to.equal('wikipedia')
+
+    if query.queryType is 'segmentMetadata'
+      expect(query.merge).to.equal(true)
+      expect(query.analysisTypes).to.be.an('array')
+
+      merged = {
+        "id": "merged",
+        "intervals": null,
+        "size": 0,
+        "numRows": 654321,
+        "columns": {
+          "__time": { "type": "LONG", "hasMultipleValues": false, "size": 0, "cardinality": null, "errorMessage": null },
```

Comment:
0.8.3 does not have `hasMultipleValues` although this is probably not affecting the usefulness of the test.